furnace.c

```diff
@@ -29,6 +29,9 @@
 
 #define MAX_TEMP 1250
 
+#define FORMAT_STATUS_FMT "temp:%d, pwm:%u/%u\n"
+#define FORMAT_STATUS_SIZE snprintf(0, 0, FORMAT_STATUS_FMT, MAX_TEMP, MAX_PWM, MAX_PWM)
```

Comment:
It's better to execute this once, at the initialization and cache the result instead of calling `snprintf` every time we need this size. Honestly It should even  be a compile time constant. Can we create a small program that generates a header file with this constant for us, before build the project? We should define a Makefile or CMake target that main program depends on, so It's run first and generate a header that the main program can include.